### PR TITLE
Add GitHub action for PHP linting

### PR DESCRIPTION
**Pull request by julmud**

### CODE REVIEW

**Comment by gjnoonan** [comment]
I will test/review this today, but my initial first impressions are that whilst  I agree that adding an action to lint is always a good idea, I tried running my usual linting tools (phpCodeSniffer, phpmd, others) before I pushed my code and there are so many style violations the added code in the MR would have been lost in the noise of the diff.

In a legacy codebase such as this there’s a couple of options;

A) A style (PSR-2 generally) is agreed or set down by the project maintainers, and then there is a big MR bringing all the code in line to this. The style is then set in stone, documented, and you have linting pre commit, merge, etc.

B) agree that new code/ features should meet the standard, but if you’re editing the old code/ files then just keep it consistent with the code around it, as bad as it may be.

In addition, if we are linting we should also add a .editorconfig in the route that defines the style for tabs/spaces, line length, and LF/CLRF

**Comment by julmud** [comment]
It was more for me a test of enabling GitHub Actions on a project. It does
just a simple lint pass (i.e., executes "php -l" on all php files), so
it'll just prevent unparseable code from landing on master.

I'll see to add an .editorconfig that matches as best as possible the
legacy codebase. As for doing more reformatting / linting, I'd say we
should stay with option B for now.

(BTW, you seem to have a way bigger knowledge of the PHP ecosystem than I
do... I'd never heard of phpCodeSniffer and phpmd before, but I hadn't done
any PHP code writing for quite a few years now ;-) )

**Comment by gjnoonan** [comment]
> It was more for me a test of enabling GitHub Actions on a project. It does just a simple lint pass (i.e., executes "php -l" on all php files), so it'll just prevent unparseable code from landing on master. 

Ah right, okay a sanity check rather than style check. Makes sense

> I'll see to add an .editorconfig that matches as best as possible the legacy codebase.

Perfect, I will review that commit.

> As for doing more reformatting / linting, I'd say we should stay with option B for now. 

Absolutely the most logical and sensible choice 👍

> (BTW, you seem to have a way bigger knowledge of the PHP ecosystem than I do... I'd never heard of phpCodeSniffer and phpmd before, but I hadn't done any PHP code writing for quite a few years now ;-) )

I wrote a lot of PHP in the late 90's/early 00's but I have not written any professionally or by choice for around 15 years. That being said although I've not written any myself, the last few years I have had to support legacy and launch new PHP applications into production for work :-)

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+root = true
+
+[*]
+charset = latin1
+end_of_line = crlf
+indent_size = 4
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[**.php]
```
Comment (gjnoonan) [inline .editorconfig:11]: It pains me to start the war, and I will keep my own opinions out of it but most of the PHP code in the codebase is indented with tabs

Now indeed the PHP community and the PSR-2/PSR-12 prefer 4 spaces. As we are putting this in to help ensure consistency, we need it to be tabs for now.

Comment (julmud) [reply .editorconfig:11]: Oops, you're absolutely right, my bad. I _did_ check the legacy files, but I guess I got distracted and wrote the wrong value.

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,9 @@
 ERROR_LOG
 DEBUG_LOG
 imagecache/
-images/
\ No newline at end of file
+images/
+
+# Prevent local collection and configuration from being uploaded to git
```
Comment (gjnoonan) [inline .gitignore:6]: Could we pull these out into a separate commit?

**Review by gjnoonan — COMMENTED** [review]
The editor config looks good to me now, just a comment about moving the giignore to a separate commit and then it should be good to merge.

File: .gitignore (hunk 2 above)
Comment (julmud) [reply .gitignore:6]: Of course I can do it. :-)
